### PR TITLE
Simplify the compilation process and the development workflow in Elm

### PR DESCRIPTION
**Pull request by jameskolce**

### CODE REVIEW

File: editor/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "start": "run-p server:** elm:**",
     "build": "npm run elm",
     "server:http": "http-server public -p 5000",
-    "server:json": "json-server --watch api/db.json --middlewares api/singular.js --port 4000 --routes api/routes.json",
+    "server:json": "json-server --watch mock/db.json --middlewares mock/singular.js --port 4000 --routes mock/routes.json",
```
Comment (jameskolce) [inline editor/package.json]: I changed it from `api` to `mock` so it's clear that it isn't a real API.

**Comment by lijunsong** [comment]
I find json-server doesn't really work well here. The biggest problem is that filesystem is really hard to  present in a RESTful way, which is how the db is structured. I could be wrong and please correct me here. I am thinking to use pollen-rock server and a real project folder when working on the front end.

**Comment by jameskolce** [comment]
I haven't checked that in detail, I just made the changes to get it working on my side in a simpler way. Today I plan to get more into it though. 

But could you explain or mention an example of why it is too hard?

**Comment by lijunsong** [comment]
This looks really clean and works really well. Let's merge it.

Now the db part, for the API of fetching file contents, list directory, etc, the resource is defined as `/fs/$PATH`, which looks not RESTful and thus is not something our db really like: to represent the resource at `path1/path/.../folder`, we'll have to write these pathes as nested objects in db, or we do the trick to convert the path to `/fs/$PATH` to `/fs?path=$PATH` and db can have one object that has a property `path: $PATH`, which still yields a giant db file that's hard to navigate and change things. The path conversion trick is the reason why there is a singular.js as middleware in the for the json-server directory. But the whole setup is funny and really complicated as a mock framework, so I didn't use that often, which in turn couldn't catch up with change of the server, so it didn't work really well here.

I am fine to keep it for now. Not sure if it would work well now when the API is more stable.

**Comment by jameskolce** [comment]
Great, and I'll check the API this afternoon, will let you know what happens!

**Comment by lijunsong** [comment]
#31 is done here.

**Pull request merged**